File: Easy/409. Longest Palindrome/Solution.java
class Solution {
    public int longestPalindrome(String s) {
        int[] letters = new int[26 * 2];
        char t;
        int result = 0;
        boolean hasOdd = false;

        for (int i = 0; i < s.length(); i++) {
            t = s.charAt(i);

            if (Character.isUpperCase(t)) {
                letters[t - 'A' + 26]++;
            } else {
                letters[t - 'a']++;
            }
        }

        for (int i = 0; i < letters.length; i++) {
            if (letters[i] % 2 == 0) {
                result += letters[i];
            } else {
                result += letters[i] - 1;
                hasOdd = true;
            }
        }
        return hasOdd ? result + 1 : result;
    }
}

File: Easy/409. Longest Palindrome/Solution.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        letters = [0] * (26 * 2)
        t = ''
        result = 0;
        hasOdd = False
        
        for char in s:
            if char.isupper():
                letters[ord(char) - ord('A') + 26] +=1
            else:
                 letters[ord(char) - ord('a')] +=1
        

        for i in letters:
            result+=i

            if i%2!=0:
                result-=1
                hasOdd = True

        return result + 1 if hasOdd else result